django_mongodb/query.py

```diff
@@ -0,0 +1,359 @@
+
+
+class MongoWhereNode:
+    """MongoDB Query Node for WhereNode"""
+
+    node_map = {
+        NothingNode: MongoNothingNode,
+        Exact: MongoExact,
+        RelatedIn: MongoRelatedIn,
+        RelatedExact: MongoExact,
+        In: MongoIn,
+        LessThan: MongoEqualityComparison,
+        LessThanOrEqual: MongoEqualityComparison,
+        GreaterThan: MongoEqualityComparison,
+        GreaterThanOrEqual: MongoEqualityComparison,
+        SearchVectorExact: MongoSearchVectorExact,
+    }
+
+    def __init__(self, where: WhereNode, mongo_meta):
+        self.node = where
+        self.connector = where.connector
+        self.children: list[MongoWhereNode | Node] = []
+        self.mongo_meta = mongo_meta
+        self.negated = where.negated
+        for child in self.node.children:
+            if isinstance(child, WhereNode):
+                self.children.append(MongoWhereNode(child, self.mongo_meta))
+            elif child.__class__ in self.node_map:
+                self.children.append(self.node_map[child.__class__](child, self.mongo_meta))
+            else:
+                raise NotImplementedError(f"Node not implemented: {type(child)}")
+
+    def __bool__(self):
+        return len(self.children) > 0
+
+    def requires_search(self) -> bool:
+        return any(child.requires_search() for child in self.children)
+
+    def get_mongo_query(self, compiler, connection, is_search=False) -> dict:
+        child_queries = list(
+            filter(
+                bool,
+                [
+                    child.get_mongo_query(compiler, connection, is_search=is_search)
+                    for child in self.children
+                ],
+            )
+        )
+        if len(child_queries) == 0:
+            return {}
+        if self.connector == "AND":
+            return {"$and": child_queries} if not self.negated else {"$nor": child_queries}
+        elif self.connector == "OR":
+            return {"$or": child_queries} if not self.negated else {"$nor": child_queries}
+        else:
+            raise Exception(f"Unsupported connector: {self.connector}")
+
+    def get_mongo_search(self, compiler, connection) -> dict:
+        child_queries = list(
+            filter(bool, [child.get_mongo_search(compiler, connection) for child in self.children])
+        )
+        if len(child_queries) == 0:
+            return {}
+        elif self.connector == "AND":
+            return {"compound": {"must": child_queries}}
+        elif self.connector == "OR":
+            return {"compound": {"should": child_queries}}
+        else:
+            raise Exception(f"Unsupported connector: {self.connector}")
+
+
+class MongoColSelect:
+    def __init__(self, col: Col, alias: str | None, mongo_meta):
+        self.col = col
+        self.mongo_meta = mongo_meta
+        self.alias = alias
+
+    def get_mongo(self):
+        return {"$project": {(self.alias or self.col.target.attname): f"${self.col.target.column}"}}
```

Comment:
This maps the MongoDB field path to the attribute names we are defining in the MongoDB model.